src/main/java/org/opensearch/security/privileges/legacy/SystemIndexAccessEvaluator.java

```diff
@@ -313,6 +313,33 @@ private void evaluateSystemIndicesAccess(
                 }
         }
 
+        // cluster actions will return true for requestedResolved.isLocalAll()
+        // the following section should only be run for index actions
+        if (this.isSystemIndexEnabled && user.isPluginUser() && !requestedResolved.isLocalAll()) {
+            Set<String> matchingSystemIndices = SystemIndexRegistry.matchesPluginSystemIndexPattern(
+                user.getName().replace("plugin:", ""),
```

Comment:
Nit here, can we add some utility function within the User class to get this information? Although User is supposed to be final....

Reply:
I added `isPluginUser` to the user class to differentiate this new type of identity corresponding to a plugin. This is used here to extract the canonical class name which is used to do a lookup in the SystemIndexRegistry which is keyed by the plugin's canonical class name.

The convention for the username for an identity corresponding to a plugin is `plugin:<canononicalClassName>`. i.e. `plugin:org.opensearch.jobscheduler.JobSchedulerPlugin`